src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -540,5 +544,25 @@ await output.Conditions.EvaluateConditionsAsync(lazyInput, context, cancellation
 
             return state;
         }
+
+        private string DefineMutexKey(Message message, Flow flow, Identity userIdentity)
+        {
+            string parent = flow.Id;
+            string identity = userIdentity.ToString();
+
+            if (message.Metadata != null && message.Metadata.ContainsKey(TUNNEL_OWNER_METADATA))
+            {
+                //when the message is traveling in the context of a router, we must consider the key as the router identifier and the tunnel originator
+                message.Metadata.TryGetValue(TUNNEL_OWNER_METADATA, out parent);
+                message.Metadata.TryGetValue(TUNNEL_ORIGINATOR_METADATA, out identity);
+            }
+            else if (!flow.ParentId.IsNullOrEmpty())
+            {
+                //when the message is traveling in the context of subflow, we must consider the key as the parent flow
+                parent = flow.ParentId;
```

Comment:
Shouldn't the key be the parent plus the id of the flow. Wouldn't this have the same issue if a user is talking with a subflow in multiple bots?

Reply:
within a conversation, the state is unique by bot...
If the user talks to the same subflow in differents bots, the key must be different too